BackEnd/src/business/playerstats.ts

```diff
@@ -58,22 +58,46 @@ export async function CreatePlayerStatsByPuuid(playerPuuid: string) {
   // Primary key is risenSeason, second key is role
   let playerStatModelMap: Map<Number, Map<String, PlayerStatModel>> = new Map<Number, Map<String, PlayerStatModel>>()
 
-  for (const playerGame of playerGames) {
-    // Check if the season exists
-    if (!playerStatModelMap.has(playerGame.seasonId)) {
-        playerStatModelMap.set(playerGame.seasonId, new Map<String, PlayerStatModel>())
+  function getRowsByRisenSeason(seasonId: number) {
+    if (!playerStatModelMap.has(seasonId)) {
+      playerStatModelMap.set(seasonId, new Map<String, PlayerStatModel>())
+    }
+
+    return playerStatModelMap.get(seasonId);
+  }
+
+  function getCurrentRow(rows: Map<String, PlayerStatModel>, lobbyPosition: string, playerGame: PlayerGameModel, seasonId: number) {
+    if(!rows.has(lobbyPosition)) {
+      rows.set(lobbyPosition, createInitialPlayerStatModel(playerGame, seasonId));
     }
 
-    let rowsByRisenSeason = playerStatModelMap.get(playerGame.seasonId);
+    return rows.get(playerGame.lobbyPosition)
+  }
 
-    // Check if the role exists
-    if(!rowsByRisenSeason.has(playerGame.lobbyPosition)) {
-       rowsByRisenSeason.set(playerGame.lobbyPosition, createInitialPlayerStatModel(playerGame));
+  function handleGame(playerGame: PlayerGameModel, seasonId: number, fullGame: GameModel) {
+    if (!playerStatModelMap.has(seasonId)) {
+      playerStatModelMap.set(seasonId, new Map<String, PlayerStatModel>())
     }
 
+    const rowsBySeasons = playerStatModelMap.get(seasonId);
+
+    if(!rowsBySeasons.has(playerGame.lobbyPosition)) {
+      rowsBySeasons.set(playerGame.lobbyPosition, createInitialPlayerStatModel(playerGame, seasonId));
+    }
+
+    const currentRow = rowsBySeasons.get(playerGame.lobbyPosition);
+    rowsBySeasons.set(playerGame.lobbyPosition, aggregateStatsForRow(currentRow, playerGame, fullGame))
+  }
+
+  for (const playerGame of playerGames) {
     const fullGame: GameModel = await GetDbGameByGameId(playerGame.gameGameId, true)
-    let currentRow: PlayerStatModel = rowsByRisenSeason.get(playerGame.lobbyPosition)
-    rowsByRisenSeason.set(playerGame.lobbyPosition, aggregateStatsForRow(currentRow, playerGame, fullGame))
+
+    handleGame(playerGame, ALL_TOURNAMENT_GAMES_ID, fullGame);
```

Comment:
Isnt this going to recursively call forever?

Reply:
This is outside the handleGame function